pgmuvi/lightcurve.py

```diff
@@ -355,13 +365,38 @@ def yerr(self):
 
     @yerr.setter
     def yerr(self, values):
+        # first, check that the input is a tensor
+        # and modifiy it if necessary
+        values = self._ensure_tensor(values)
+        # then, store the raw data internally
         self._yerr_raw = values
         # now apply the same transformation that was applied to the ydata
         if self.ytransform is None:
             self._yerr_transformed = values
         elif isinstance(self.ytransform, Transformer):
             self._yerr_transformed = self.ytransform.transform(values)
 
+    def _ensure_tensor(self, values):
```

Comment:
In principle, I think this could be moved outside the context of the class, as maybe it would make sense as a more general utility? But I think this is great for now, and we can think about that later.

Reply:
I'd say that this method could stay as an internal method as I don't see any other utility for it outside of the scope of making sure the data is in the correct form. Now that being said, we could indeed imagine a more general method to transform data in the right type. I would then follow your last sentence.